.github/workflows/perf-check.yml

```diff
@@ -0,0 +1,61 @@
+# Adapted from ember-data https://github.com/emberjs/data/blob/master/.github/workflows/perf-check.yml
+name: Performance Checks
+
+on:
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  performance-checks:
+    name: 'Performance Checks'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Check SHA
+        run: |
+          # Github actions have had a race condition where sometimes they don't check out the correct sha
+          # Here we log our starting state in case we run into that again
+          sha=$(git rev-parse --short=8 HEAD)
+          echo "HEAD sha=$sha"
+          echo "HEAD sha=$GITHUB_SHA"
+          echo "GITHUB_SHA sha=$GITHUB_SHA"
+          mkdir -p tmp
+          echo $sha > tmp/sha-for-check.txt
+          originSha=$(git rev-parse HEAD^2)
+          echo $originSha > tmp/sha-for-commit.txt
+          git show --format=short --no-patch $originSha
+      - uses: tracerbench/tracerbench-compare-action@master
+        with:
+          experiment-build-command: cd m3-perf-testing-app && rm -rf node_modules && yarn && ./node_modules/ember-cli/bin/ember build -e production --output-path dist-experiment
+          experiment-serve-command:  cd m3-perf-testing-app &&  ./node_modules/ember-cli/bin/ember s --path dist-experiment --port 4201
+          control-build-command:  cd m3-perf-testing-app && rm -rf node_modules &&  yarn &&  ./node_modules/ember-cli/bin/ember  build -e production --output-path dist-control
+          control-serve-command:  cd m3-perf-testing-app &&  ./node_modules/ember-cli/bin/ember s --path dist-control
+          scenarios: |
+            {
+              "materializing": {
+                "control": "http://localhost:4200/materializing",
+                "experiment": "http://localhost:4201/materializing",
+                "markers": "start-loading,pushed-payload,end-loading"
+              },
+              "rendering": {
+                "control": "http://localhost:4200/rendering",
+                "experiment": "http://localhost:4201/rendering",
+                "markers": "start-loading,pushed-payload,end-loading"
+              }
+            }
+          fidelity: 60
+          upload-traces: true
+          upload-results: true
+      - name: Report TracerBench Results
+        if: failure() || success()
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Just to be clear, the usage of `${{ secrets.* }}` here means that this job will only run on PRs that are _from_ a branch on `hjdivad`'s repo (e.g. not from forks). Is that correct?

Reply:
There isn't a way to do the report step I believe without the token. Is there an alternative approach you would suggest?

Reply:
We can't do the comment step, but we definitely could run TracerBench itself and have it pass/fail the build.

That way the comment would be additive / nice to have, but we still have the same fundamental feature set either way.

Reply:
Going through github docs, it seems that it can be set to be available but only in a read only mode, which will make the job run, but the comment to fail.  https://docs.github.com/en/actions/reference/authentication-in-a-workflow#permissions-for-the-github_token

Thats what seems to happen for ember data as well, where for outside PRs, the perf runs but the comment fails to post

https://github.com/emberjs/data/pull/7607/checks?check_run_id=2891069546#step:5:14

Reply:
Ya, so we'll need @hjdivad to set that up (since it's a personal repo, only he is an admin).

> Thats what seems to happen for ember data as well, where for outside PRs, the perf runs but the comment fails to post

Overall, this is fine, but I want to ensure that we have a reasonable threshold set that will fail the perf-check CI job (make the status line red, to force an admin to look at the results).

Reply:
Yup, I am going to watch the next few PRs to see what the current noise looks like, and then consult with @hjdivad once he is back